discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/01-setting-up-identity-provider.markdown

```diff
@@ -0,0 +1,271 @@
+service provider, but **not both**. Both configurations are covered in this
+article.
+
+To set @product@ up to act as a SAML Identity Provider, follow the steps below.
+Before proceeding, note that in step 3 below, you generate a keystore for SAML.
+This keystore has two storage options:
+
+    - In the file system
+    - In the Documents and Media library
+ 
+The file system keystore manager is used by default and the default location is
+the `[Liferay Home]/data` directory. To use Documents and Media library storage
+for your keystore instead of file system storage, use the document library
+keystore manager.
+
+To select a keystore manager, to to *Control Panel* &rarr; *System Settings*
+&rarr; *SAML KeyStoreManager Implementation Configuration*. There, the options
+are *Filesystem Keystore Manager* and *Document Library Keystore Manager*.
+
+The great thing about using Document Library storage is that you can use any number
+of [back end file stores](/discover/deployment/-/knowledge_base/7-0/document-repository-configuration).
+These are protected not only by the system in which you're storing the key, but
+also by @product@'s permissions system.
+
+Here are the steps for setting up @product@ to act as a SAML Identity Provider:
+
+1.  Install the Liferay SAML 2.0 Provider app. To access the SAML Admin
+    interface, click on *Control Panel* &rarr; *Configuration* and then on *SAML Admin*.
+
+2.  To begin configuring @product@ to use SAML, select a SAML role for @product@ and
+    choose an entity ID.
+
+    ![Figure 1: Select a SAML role for Liferay and enter an entity ID.](../../../../images-dxp/saml-initial-config.png)
+
+    Select the *Identity Provider* SAML role. Enter *liferaysamlidp* if you're
+    setting up an example @product@ instance. Alternatively, choose your own
+	entity ID. Then click *Save*. A new Certificate and Private Key section
+	appears. 
+
+3.  The Certificate and Private Key section lets you create a keystore for SAML.
+    Click on *Create Certificate* and enter the following information:
+    
+    - Your common name (your first and last name)
+    - The name of your organization
+    - The name of your organizational unit
+    - The name of your city or locality
+    - The name of your state or province
+    - The name of your country
+    - The length in days that your keystore will remain valid (how long before
+      the keystore expires)
+    - The key algorithm (RSA is the default)
+    - The key length in bits (2048 is the default)
+    - The key password
+
+	If you're making a test configuration, use the password *liferay*. When
+    you enter all the required information, click *Save*.
+
+	Note that the SAML keystore is created by the SAML app's keystore
+	manager. As described above, this keystore has two storage options, file
+	system storage (the default) and documents and media storage. 
```

Comment:
Maybe mention that both a keystore and certificate + private key entry is created.
Also, the keystore is only created if it does not already exist, otherwise the existing one is used to store the new certificate + private key entry, so long as @Product is configured with the correct "Key Store password" in Control Panel → System Settings → SAML Configuration.

Reply:
I was assuming that readers would know the difference between the keystore and what it contains, but maybe that was a bad assumption: I can clarify.

Reply:
That might be a reasonable assumption. I'm thinking that it by separating our the concepts it becomes easier later to talk about scenarios such as adding a CA issued certificate to a generated keystore using the offline keytool